### PR TITLE
Move `@storybook/addons` to peerDependencies in all addons

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #1981 #1892 
We need a single instance of `@storybook/addons` to be shared between addons and app.

## What I did
I moved the package to peerDeps in all the addons (some of them already had it there). There's no way to use addons without an app, so users still don’t have to install addons package on their side. Note that this relies on npm flattening the dependencies (the addons package is kept as a direct dep in apps)

A better solution would be to make the package a peerDep everywhere including the app, but that would be a breaking change, as users would have to install the package on their side.

## How to test

I actually can’t find a good way to test that without publishing a new alpha version =(

### CODE REVIEW

File: addons/info/package.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
     "storybook": "start-storybook -p 9010"
   },
   "dependencies": {
-    "@storybook/addons": "^3.3.0-alpha.3",
```
Comment (Hypnosphi) [inline addons/info/package.json]: It isn't actually used here

**Review by TroySchmidt — CHANGES_REQUESTED** [review]
Get it up to date with the latest alpha release all the way thru to push this out.

File: addons/links/package.json [hunk 2]
```diff
@@ -33,6 +32,7 @@
     "shelljs": "^0.7.8"
   },
   "peerDependencies": {
+    "@storybook/addons": "^3.3.0-alpha",
```
Comment (ndelangen) [inline addons/links/package.json]: should this be `^3.3.0-alpha.3` ?

Comment (Hypnosphi) [reply addons/links/package.json]: It depends on whether we want to allow different versions of our packages to be used. If not, we should omit the caret in all cross-dependencies (direct and peer)

Comment (Hypnosphi) [reply addons/links/package.json]: Ok, it turns out that lerna can update peerDeps as well, so I changed the version to the current one everywhere

**Review by ndelangen — APPROVED** [review]

**Comment by TroySchmidt** [comment]
Cannot wait to try this out!

**Comment by danielduan** [comment]
@Hypnosphi do you know what happened to this change? looks like it got lost in the latest master?

**Pull request merged**